packages/core/src/App/Containers/RealAccountSignup/account-wizard.jsx

```diff
@@ -69,42 +69,41 @@ class AccountWizard extends React.Component {
         this.props.fetchStatesList();
         const { cancel, promise } = makeCancellablePromise(this.props.fetchResidenceList());
         this.cancel = cancel;
+        const { cancel: cancelFinancialAssessment, promise: financial_assessment_promise } = makeCancellablePromise(
+            this.props.fetchFinancialAssessment()
+        );
+        this.cancelFinancialAssessment = cancelFinancialAssessment;
+
+        Promise.all([promise, financial_assessment_promise]).then(() => {
+            this.setState({
+                items: getItems(this.props),
+                mounted: false,
+            });
 
-        promise
-            .then(() => {
+            if (!this.residence_list?.length) {
+                const items = this.state.items.slice(0);
```

Comment:
line 84-88 is a bit unreadable to me 😅 , can you help to add comment on what happen if residence list is empty 🙏

Reply:
```suggestion
                const items = [...this.state.items];
```